addons/point_of_sale/static/src/js/db.js

```diff
@@ -403,7 +403,7 @@ var PosDB = core.Class.extend({
         if (product_ids) {
             for (var i = 0, len = Math.min(product_ids.length, this.limit); i < len; i++) {
                 const product = this.product_by_id[product_ids[i]];
-                if (!(product.active && product.available_in_pos)) continue;
+                if (!(product.show_product_in_pos)) continue;
```

Comment:
I guess you'll have to extract the method in js code because adding new field in the model is not allowed in stable. So you can do something like:
```suggestion
                if (!(product.shouldDisplay())) continue;
```

And then in Product class (from models.js), define the `shouldDisplay` method.

Then in your module, you can patch the Product class.